packages/swing-store/src/swingStore.js

```diff
@@ -623,6 +851,180 @@ export function initSwingStore(dirPath = null, options = {}) {
   return makeSwingStore(dirPath, true, options);
 }
 
+function parseExportKey(key) {
+  const parts = key.split('.');
+  const [tag, type, vatID, rawPos] = parts;
+  assert(
+    parts.length === 4 && tag === 'export',
+    `expected artifact name of the form 'export.{type}.{vatID}.{pos}', saw ${key}`,
+  );
+  assert(
+    type === 'snapshot' || type === 'transcript',
+    `invalid artifact type ${type}`,
+  );
+  const isCurrent = rawPos === 'current';
+  let pos;
+  if (isCurrent) {
+    pos = -1;
+  } else {
+    pos = Number(rawPos);
+  }
+
+  return { type, vatID, isCurrent, pos };
+}
+
+function artifactKey(type, vatID, pos) {
+  return `${type}.${vatID}.${pos}`;
+}
+
+/**
+ * @param {SwingStoreExporter} exporter
+ * @param {string | null} [dirPath]
+ * @param {object} options
+ * @returns {Promise<SwingStore>}
+ */
+export async function importSwingStore(exporter, dirPath = null, options = {}) {
+  if (dirPath) {
+    assert.typeof(dirPath, 'string');
+  }
+  const { includeHistorical = false } = options;
+  const store = makeSwingStore(dirPath, true, options);
+  const { kernelStorage } = store;
+
+  // Artifact metadata, keyed as `${type}.${vatID}.${pos}`
+  //
+  // Note that this key is almost but not quite the artifact name, since the
+  // names of transcript span artifacts also include the endPos, but the endPos
+  // value is in flux until the span is complete.
+  const artifactMetadata = new Map();
+
+  // Each vat requires a transcript span and (usually) a snapshot.  This table
+  // tracks which of these we've seen, keyed by vatID.
+  // vatID -> { snapshotKey: metadataKey, transcriptKey: metatdataKey }
+  const vatArtifacts = new Map();
+
+  for (const [key, value] of exporter.getKVData()) {
+    if (key.startsWith('export.')) {
+      // Keys starting with 'export.' contain artifact description info,
+      // intended for us here in the import stage rather than items destined for
+      // the KV store.
+      assert(value); // make TypeScript shut up
+      const { type, vatID, isCurrent, pos } = parseExportKey(key);
+      if (isCurrent) {
+        const vatInfo = vatArtifacts.get(vatID) || {};
+        if (type === 'snapshot') {
+          // `export.snapshot.{vatID}.current` directly identifies the current snapshot artifact
+          vatInfo.snapshotKey = value;
+        } else if (type === 'transcript') {
+          // `export.transcript.${vatID}.current` contains a metadata record for the current
+          // state of the current transcript span as of the time of export
+          const metadata = JSON.parse(value);
+          vatInfo.transcriptKey = artifactKey(type, vatID, metadata.startPos);
+          artifactMetadata.set(vatInfo.transcriptKey, metadata);
+        }
+        vatArtifacts.set(vatID, vatInfo);
+      } else {
+        artifactMetadata.set(artifactKey(type, vatID, pos), JSON.parse(value));
```

Comment:
That is not a thing that can happen as `export.transcript.${vatID}.${startPos}` only describes historical spans and  `export.transcript.${vatID}.current` only describes the current span.

Reply:
Ah interesting, I missed that. I'll need to look again at how the `.current` entries are used/generated, but it might be a sign that some more comments are needed.